### ISSUE TITLE
WARNING:  Unsupported upgrade request.

### ISSUE THREAD

**Issue by ghost** [tiangolo/fastapi#2209]
```py
app = FastAPI()

@app.websocket("/ws")
async def websoc(websocket: WebSocket):
    await websocket.accept()
    await websocket.send_json({"msg": "Hello WebSocket"})
    await websocket.close()
```

When attempting to run this (using UviCorn), it starts fine

then, when i try and use my client to connect the the websocket, i get this error

```bash
Python 3.8.6 (tags/v3.8.6:db45529, Sep 23 2020, 15:52:53) [MSC v.1927 64 bit (AMD64)] on win32
>>> runfile('C:/Users/Tobi/Documents/Scripting/Mixed/Webpage/main.py', wdir='C:/Users/Tobi/Documents/Scripting/Mixed/Webpage')
INFO:     Started server process [9580]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://192.168.1.88:80 (Press CTRL+C to quit)
**WARNING:  Unsupported upgrade request.**
```

This is my client code, its just supposed to be a very simple test

```py
import websocket, time

ws = websocket.create_connection("ws://192.168.1.88/ws")

while True:
    ws.recv()
    time.sleep(2)
```

Let me know if you can help, or if anyone else has had similar issues with this.

**Comment by eseglem** [comment]
This error is not part of the FastAPI codebase. It appears to come from [uvicorn](https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/h11_impl.py#L263) and `fastapi.WebSocket` is a wrapper around [`starlette.websockets.WebSocket`](https://www.starlette.io/websockets/)

How are you calling `uvicorn` for your app? 

Perhaps: [uvicorn/test_websocket.py](https://github.com/encode/uvicorn/blob/master/tests/protocols/test_websocket.py) or [starlette/test_websocket.py](https://github.com/encode/starlette/blob/master/tests/test_websockets.py) might be a good place to look to see how they are configuring the tests.

**Comment by TobiWtf** [comment]
> This error is not part of the FastAPI codebase. It appears to come from [uvicorn](https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/h11_impl.py#L263) and `fastapi.WebSocket` is a wrapper around [`starlette.websockets.WebSocket`](https://www.starlette.io/websockets/)
> 
> How are you calling `uvicorn` for your app?
> 
> Perhaps: [uvicorn/test_websocket.py](https://github.com/encode/uvicorn/blob/master/tests/protocols/test_websocket.py) or [starlette/test_websocket.py](https://github.com/encode/starlette/blob/master/tests/test_websockets.py) might be a good place to look to see how they are configuring the tests.

I am using uvicorn, however, I'm also running an HTTP server with fast api along with the Websocket, is it possible to make those work on the same port using uvicorn?

**Comment by TobiWtf** [comment]
> This error is not part of the FastAPI codebase. It appears to come from [uvicorn](https://github.com/encode/uvicorn/blob/master/uvicorn/protocols/http/h11_impl.py#L263) and `fastapi.WebSocket` is a wrapper around [`starlette.websockets.WebSocket`](https://www.starlette.io/websockets/)
> 
> How are you calling `uvicorn` for your app?
> 
> Perhaps: [uvicorn/test_websocket.py](https://github.com/encode/uvicorn/blob/master/tests/protocols/test_websocket.py) or [starlette/test_websocket.py](https://github.com/encode/starlette/blob/master/tests/test_websockets.py) might be a good place to look to see how they are configuring the tests.

As suspected, this was a uvicorn issue, not FastAPI, however, to fix this, for anyone else curious, goto terminal and type

```bash
python3 -m pip uninstall uvicorn
```

then

```bash
python3 -m pip install uvicorn[standard]
```

The original uvicorn package doesnt come with websocket support, you have to download the standard one.

**Comment by KarlCMB** [comment]
Had the same issue.
```
pip install websockets
```

did the job

**Comment by wiltonsr** [comment]
The [docs](https://fastapi.tiangolo.com/advanced/websockets/) should mention that.

**Comment by 2114L3** [comment]
https://github.com/encode/uvicorn/releases/tag/0.12.0 mentions https://github.com/encode/uvicorn/pull/666 
this is an issue that should be handled here https://github.com/tiangolo/uvicorn-gunicorn-docker

**Comment by BassdoxXx** [comment]
I've received the same issue.
I just fixed by cancel the current running uvicorn server and start it again.

**Comment by adriangabura** [comment]
Jesus, please add to the docs this thing regarding uvicorn[standard] websockets dependencies. I spend literally 2 hours going in circles until I've seen this thread! Thanks to the guys who suggested that!

**Comment by friendlyemu** [comment]
If anyone was wondering, I couldn't get this working using Python 3.9. Needed to downgrade to specifically 3.8 to fix it.

**Comment by TobiWtf** [comment]
> Jesus, please add to the docs this thing regarding uvicorn[standard] websockets dependencies. I spend literally 2 hours going in circles until I've seen this thread! Thanks to the guys who suggested that!

Yup, should be covered in the docs but it's not. Took be days to figure out.

**Comment by ahmetonurslmz** [comment]
> Had the same issue.
> 
> ```
> pip install websockets
> ```
> 
> did the job

really nice answer to solve websockets issue. Works now

**Comment by tiangolo** [comment]
Thanks for the help here everyone! 👏  🙇 

Thanks for reporting back and closing the issue @baiinss  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
